### PR TITLE
[TILES-COLLAB-7] PLU-238: hide/show certain functions depending on role

### PR DESCRIPTION
**Pull request by pregnantboy**
## Changes

Referencing this: https://www.notion.so/opengov/Tiles-Collaborator-50eba0b1019b4b5c8de1355bf97d7784?pvs=4

  | Owner | Editor | Viewer | Shareable link
-- | -- | -- | -- | --
Can view tile | ✅ | ✅ | ✅ | ✅
Can view collaborators | ✅ | ✅ | ✅ | ✅
Can view shareable link | ✅ | ✅ | ✅ | ✅
Can edit data | ✅ | ✅ |   |  
Can adjust/edit columns | ✅ | ✅ |   |  
Can modify collaborators | ✅ | ✅ |   |  
Can regenerate shareable link | ✅ | ✅ |   |  
Can use in pipe | ✅ |   |   |  
Can delete table | ✅ |   |   |  

This PR hides/disables certain features based on this permission matrix

### CODE REVIEW

**Comment by pregnantboy** [comment]
* **#603** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/603?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#602** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/602?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#601** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/601?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#600** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/600?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#599** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/599?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#597** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/597?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pregnantboy and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/pages/Tile/components/TableBanner/TableCollaborators.tsx [hunk 1]
```diff
@@ -73,7 +73,7 @@ const AddNewCollaborator = ({
   const onSubmit: FormEventHandler<HTMLFormElement> = useCallback(
     async (e) => {
       try {
-        e.preventDefault()
+        e.stopPropagation()
```
Comment (m0nggh) [inline packages/frontend/src/pages/Tile/components/TableBanner/TableCollaborators.tsx]: question: can i ask why you change to `stopPropagation` ah

File: packages/frontend/src/pages/Tile/components/TableBanner/ShareButton.tsx [hunk 2]
```diff
@@ -67,54 +67,63 @@ const ShareModal = ({ onClose }: { onClose: () => void }) => {
...
-              {viewOnlyKey && (
-                <FormHelperText variant={isNewLink ? 'success' : undefined}>
-                  {isNewLink
-                    ? 'New link generated!'
-                    : 'By generating a new link, your previous link will not work anymore.'}
-                </FormHelperText>
-              )}
-            </VStack>
-            <Divider my={6} />
+            {(viewOnlyKey || hasEditPermission) && (
```
Comment (m0nggh) [inline packages/frontend/src/pages/Tile/components/TableBanner/ShareButton.tsx:70]: sorz i a lil confused when would this not be true ah

File: packages/frontend/src/pages/Tile/components/TableBanner/EditMode.tsx [hunk 3]
```diff
@@ -36,14 +36,14 @@ const EditMode = () => {
     [mode],
   )
 
-  if (mode === 'edit') {
+  if (hasEditPermission) {
```
Comment (m0nggh) [inline packages/frontend/src/pages/Tile/components/TableBanner/EditMode.tsx:39]: actly why you dont want to show the edit mode badge ah

**Review by m0nggh — APPROVED** [review]
other than some questions, tested that the permission matrix works based on the different roles, good job with the clean coding, lgtm

File: packages/frontend/src/pages/Tile/components/TableBanner/TableCollaborators.tsx (hunk 1 above)
Comment (pregnantboy) [reply packages/frontend/src/pages/Tile/components/TableBanner/TableCollaborators.tsx]: oops i forgot to change back

File: packages/frontend/src/pages/Tile/components/TableBanner/ShareButton.tsx (hunk 2 above)
Comment (pregnantboy) [reply packages/frontend/src/pages/Tile/components/TableBanner/ShareButton.tsx:70]: when shareable link has not been generated and it's a view only state

File: packages/frontend/src/pages/Tile/components/TableBanner/EditMode.tsx (hunk 3 above)
Comment (pregnantboy) [reply packages/frontend/src/pages/Tile/components/TableBanner/EditMode.tsx:39]: designer choice

**Pull request merged**